tests/components/weatherflow/test_config_flow.py

```diff
@@ -0,0 +1,104 @@
+"""Tests for WeatherFlow."""
+
+import asyncio
+from unittest.mock import AsyncMock, patch
+
+from pyweatherflowudp.errors import AddressInUseError
+
+from homeassistant import config_entries
+from homeassistant.components.weatherflow.const import (
+    DOMAIN,
+    ERROR_MSG_ADDRESS_IN_USE,
+    ERROR_MSG_CANNOT_CONNECT,
+    ERROR_MSG_NO_DEVICE_FOUND,
+)
+from homeassistant.core import HomeAssistant
+from homeassistant.data_entry_flow import FlowResultType
+
+from tests.common import MockConfigEntry
+
+
+async def test_single_instance(
+    hass: HomeAssistant,
+    mock_config_entry: MockConfigEntry,
+    mock_has_devices: AsyncMock,
+) -> None:
+    """Test more than one instance."""
+    mock_config_entry.add_to_hass(hass)
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN,
+        context={"source": config_entries.SOURCE_USER},
+    )
+    assert result["type"] == FlowResultType.ABORT
+    assert result["reason"] == "single_instance_allowed"
+
+
+async def test_devices_with_mocks(
+    hass: HomeAssistant, mock_start: AsyncMock, mock_stop: AsyncMock
+) -> None:
+    """Test getting user input."""
+
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN,
+        context={"source": config_entries.SOURCE_USER},
+    )
+    await hass.async_block_till_done()
+    assert result["type"] == FlowResultType.CREATE_ENTRY
+    assert result["data"] == {}
+
+
+async def test_devices_with_various_mocks_errors(
```

Comment:
Please use `@pytest.mark.parametrize` for each exception and the expected error

Reply:
-> And I still need to create or abort to close it out right?